### DISCUSSION TITLE
Is it possible to use shell integration without losing my posh-git prompts?

### DISCUSSION THREAD

**Discussion by jnm2** [microsoft/terminal#16893] [Q&A]
I'm enabling shell integration (<https://learn.microsoft.com/en-us/windows/terminal/tutorials/shell-integration#powershell-pwshexe>) because I wanted the Duplicate Tab feature to open a new tab with the same working directory that I was just in (<https://learn.microsoft.com/en-us/windows/terminal/tutorials/shell-integration#open-new-tabs-in-the-same-working-directory>).

However, my [posh-git prompt](https://github.com/dahlbyk/posh-git?tab=readme-ov-file#overview) is now gone. Is there a way to get them to work together?

**Comment by zadjii-msft** [comment]
Just looking at the docs on posh-git, I'd say it's doable. 

Looks like the relevant doc section is
https://github.com/dahlbyk/posh-git#gitpromptscriptblock

Absolutely spitballing, "I have not tested this", but I'd guess:

```pwsh
# my profile.ps1
function prompt {
    # Your non-prompt logic here
    Write-Host "`e]9;9;`"$loc`"$([char]07)" -nonewline
    $prompt = & $GitPromptScriptBlock

    if ($prompt) { "$prompt " } else { " " }
}
```

Should have powershell emit the CWD before the posh-git prompt.
